lib/clients/aws-client/lambda-client.js

```diff
@@ -16,24 +15,25 @@ module.exports = class LambdaClient extends AbstractAwsClient {
      * Wrapper of aws sdk api
      * Creates a Lambda function
      * @param {string} functionName The name of the Lambda function.
+     * @param {object} config The configuration of the Lambda function.
      * @param {string} role The Amazon Resource Name (ARN) of the function's execution role.
      * @param {Buffer} zipFile The base64-encoded contents of the deployment package.
-     * @param {string} runtime The identifier of the function's runtime.
-     * @param {string} handler The name of the method within your code that Lambda calls to execute your function.
      * @param {callback} callback { error, response }
      */
-    createLambdaFunction(skillName, profile, alexaRegion, role, zipFile, runtime, handler, callback) {
+    createLambdaFunction(functionName, config, role, zipFile, callback) {
+        const { runtime, handler, description, memorySize, timeout, environmentVariables } = config;
         const params = {
             Code: {
                 ZipFile: zipFile
             },
+            FunctionName: functionName,
             Role: role,
             Runtime: runtime,
             Handler: handler,
-            // to account for Java runtime cold start
-            Timeout: 15,
-            MemorySize: 512,
-            FunctionName: this._generateFunctionName(skillName, profile, alexaRegion)
+            ...(description && { Description: description }),
+            MemorySize: memorySize,
+            Timeout: timeout,
+            ...(environmentVariables && { Environment: { Variables: environmentVariables } })
```

Comment:
nit: I think keeping the optional/possibly undefined stuff would make this object a little bit cleaner

```
const params = {
   Code: {
                ZipFile: zipFile
    },
     FunctionName: functionName,
     Handler: handler,
     MemorySize: memorySize,
     Role: role,
     Runtime: runtime,
     Timeout: timeout,
     ...(description && { Description: description }),
     ...(environmentVariables && { Environment: { Variables: environmentVariables } 
})
```